tests/Functional/CorporateFooterRenderTest.php

```diff
@@ -0,0 +1,435 @@
+    $this->assertEquals('Drupal', $actual->getText());
+    $this->assertEquals('http://web:8080/build/', $actual->getAttribute('href'));
+
+    // Site owner is not set yet, lets make sure we don't have a description.
+    $actual = $assert->elementExists('css', 'div.ecl-footer-standardised__description');
+    $this->assertEquals('Discover more on europa.eu', $actual->getText());
+
+    $section = $assert->elementExists('css', 'footer.ecl-footer-standardised section.ecl-footer-standardised__section7');
+
+    // Assert presence of ecl logo in footer.
+    $this->assertEclLogoPresence($section, 'standardised');
+
+    $section = $assert->elementExists('css', 'footer.ecl-footer-standardised section.ecl-footer-standardised__section8 .ecl-footer-standardised__section:nth-child(1)');
+
+    $actual = $section->find('css', '.ecl-footer-standardised__title');
+    $this->assertEquals('Contact title', $actual->getText());
+
+    $items = $data['contact'];
+
+    foreach ($items as $key => $expected) {
+      $index = $key + 1;
+      $actual = $section->find('css', "ul li:nth-child({$index}) > a");
+      $this->assertListLink($actual, 'standardised', $expected);
+    }
+
+    $section = $assert->elementExists('css', 'footer.ecl-footer-standardised section.ecl-footer-standardised__section8 .ecl-footer-standardised__section:nth-child(2)');
+
+    $actual = $section->find('css', '.ecl-footer-standardised__title');
+    $this->assertEquals('Social media title', $actual->getText());
+
+    $items = $data['social_media'];
+
+    foreach ($items as $key => $expected) {
+      $index = $key + 1;
+      $actual = $section->find('css', "ul li:nth-child({$index}) > a");
+      $this->assertListLink($actual, 'standardised', $expected);
+    }
+
+    $section = $assert->elementExists('css', 'footer.ecl-footer-standardised section.ecl-footer-standardised__section8 .ecl-footer-standardised__section:nth-child(3)');
+
+    $actual = $section->find('css', '.ecl-footer-standardised__title');
+    $this->assertEquals('Legal links title', $actual->getText());
+
+    $items = $data['legal_links'];
+
+    foreach ($items as $key => $expected) {
+      $index = $key + 1;
+      $actual = $section->find('css', "ul li:nth-child({$index}) > a");
+      $this->assertListLink($actual, 'standardised', $expected);
+    }
+
+    $section = $assert->elementExists('css', 'footer.ecl-footer-standardised section.ecl-footer-standardised__section9');
+
+    $actual = $section->find('css', '.ecl-footer-standardised__title');
+    $this->assertEquals('Institution links title', $actual->getText());
+
+    $items = $data['institution_links'];
+
+    foreach ($items as $key => $expected) {
+      $index = $key + 1;
+      $actual = $section->find('css', "ul li:nth-child({$index}) > a");
+      $this->assertListLink($actual, 'standardised', $expected);
+    }
+
+    // Update settings, assert footer changed.
+    $this->updateSiteSettings();
```

Comment:
This is similar to the issue we had outlined in a previous review: the caller needs to pass to the method the values to set, even if it's used once, or else the caller needs to know what that method does.

Change this so that we can set site owner and name explicitly and assert them here below.